src/main/java/net/neoforged/neoforge/registries/NeoForgeRegistries.java

```diff
@@ -36,7 +36,7 @@ public class NeoForgeRegistries {
     public static final Registry<MapCodec<? extends BiomeModifier>> BIOME_MODIFIER_SERIALIZERS = new RegistryBuilder<>(Keys.BIOME_MODIFIER_SERIALIZERS).create();
     public static final Registry<MapCodec<? extends StructureModifier>> STRUCTURE_MODIFIER_SERIALIZERS = new RegistryBuilder<>(Keys.STRUCTURE_MODIFIER_SERIALIZERS).create();
     public static final Registry<FluidType> FLUID_TYPES = new RegistryBuilder<>(Keys.FLUID_TYPES).sync(true).create();
-    public static final Registry<HolderSetType> HOLDER_SET_TYPES = new RegistryBuilder<>(Keys.HOLDER_SET_TYPES).create();
+    public static final Registry<HolderSetType> HOLDER_SET_TYPES = new RegistryBuilder<>(Keys.HOLDER_SET_TYPES).sync(true).create();
```

Comment:
This registry needs to have a max id, since we are syncing the id over the space limited by half a `VarInt`.

It's also worth noting the reason for the max id here, so people don't have to go into the network code to figure it out.

Reply:
Nope, `VarInt` has `5 * 7 = 35` bits for data and thus could hold a whole int as the name stated, the actual value we are using is [`Integer.MIN_VALUE`, -1], which happens to be of the same size as [0, `Integer.MAX_VALUE`] (the max id range for a registry).